### PR TITLE
Core/Misc: Ensure that all actions are compared to fixed point in time (ie. world update start)

### PR DESCRIPTION
**Pull request by Viste**
##### CHANGES PROPOSED:
- Actions will not be dependent on processing moment
- Increased GameObjects cooldown resolution to milliseconds, fixes arming time of traps to be exactly one second and not something from range (1000, 1999)
- Created GameTime namespace and UpdateTime class and moved there some code out of world
##### HOW TO TEST THE CHANGES: all should work as now
**In General need test**:

- [ ] Battlegrounds
- [ ] Arenas
- [ ] ICC
- [ ] Ulduar
- [ ] Wintergrasp and VoA
- [ ] brewfest
- [ ] midsummer
- [ ] spells
- [ ] .server commands
- [ ] .ban commands

##### Target branch(es):Master

### CODE REVIEW

**Comment by FrancescoBorzi** [comment]
is this importing from TC?

**Comment by Viste** [comment]
@FrancescoBorzi yes

File: src/server/game/Chat/Channels/Channel.h [hunk 1]
```diff
@@ -164,9 +164,9 @@ class Channel
         }
         bool IsAllowedToSpeak(uint32 speakDelay) // pussywizard
         {
-            if (lastSpeakTime+speakDelay <= sWorld->GetGameTime())
+            if (lastSpeakTime+speakDelay <= GameTime::GetGameTime())
```
Comment (Nefertumm) [inline src/server/game/Chat/Channels/Channel.h]: Missing include in this file?

Comment (Viste) [reply src/server/game/Chat/Channels/Channel.h]: yes and in many others )

**Comment by Viste** [comment]
redy to test

**Comment by Viste** [comment]
In General need test:
Battlegrounds
Arenas
ICC
Ulduar
Wintergrasp and VoA
brewfest 
midsummer
spells

**Comment by BarbzYHOOL** [comment]
Hard to test for non devs no? (i mean I don't even see how to test this)

**Comment by Viste** [comment]
I'm worried, for example, for the mute(изменено)
and .server info command
bg start and end time
gunship battle timers

**Comment by FrancescoBorzi** [comment]
Merged in Staging since contains generic changes that are hard to test properly and might affect a wide range of things.

Staging will be merged into master after some testing with test players.

**Comment by kvipka** [comment]
https://at-wow.org/info/65-test-real-server-diff-of-clear-azerothcore-and-overloaded-atieshcore-trinitycore.html

**Comment by Nefertumm** [comment]
Congrats. What are you trying to achieve by that?

**Comment by FrancescoBorzi** [comment]
@kvipka good job with creating a private fork of TrinityCore and breaking its GPL license

**Pull request merged**